File: HRMS/HRMS.Web/Modules/Recruitment/Candidates/CandidatesForm.cs
namespace HRMS.Recruitment.Forms;

[FormScript("Recruitment.Candidates")]
[BasedOnRow(typeof(CandidatesRow), CheckNames = true)]
public class CandidatesForm
{
    [LookupEditor(typeof(JobOpeningsRow))]
    public int JobId { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string Email { get; set; }
    public string Mobile { get; set; }
    public string Resume { get; set; }
    public CandidateStatus Status { get; set; }
    public DateTime AppliedDate { get; set; }
}

File: HRMS/HRMS.Web/Modules/ServerTypes/Recruitment/CandidatesForm.ts
import { DateEditor, EnumEditor, ImageUploadEditor, initFormType, LookupEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";
import { CandidateStatus } from "./CandidateStatus";

export interface CandidatesForm {
    JobId: LookupEditor;
    FirstName: StringEditor;
    LastName: StringEditor;
    Email: StringEditor;
    Mobile: StringEditor;
    Resume: ImageUploadEditor;
    Status: EnumEditor;
    AppliedDate: DateEditor;
}

export class CandidatesForm extends PrefixedContext {
    static readonly formKey = 'Recruitment.Candidates';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CandidatesForm.init) {
            CandidatesForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;
            var w2 = ImageUploadEditor;
            var w3 = EnumEditor;
            var w4 = DateEditor;

            initFormType(CandidatesForm, [
                'JobId', w0,
                'FirstName', w1,
                'LastName', w1,
                'Email', w1,
                'Mobile', w1,
                'Resume', w2,
                'Status', w3,
                'AppliedDate', w4
            ]);
        }
    }
}

[CandidateStatus]; // referenced types